cockatrice/src/chatview/chatview.cpp

```diff
@@ -145,76 +152,86 @@ void ChatView::appendUrlTag(QTextCursor &cursor, QString url)
 
 void ChatView::appendMessage(QString message,
                              RoomMessageTypeFlags messageType,
-                             QString sender,
+                             const QString &userName,
                              UserLevelFlags userLevel,
                              QString UserPrivLevel,
                              bool playerBold)
 {
     bool atBottom = verticalScrollBar()->value() >= verticalScrollBar()->maximum();
-    bool sameSender = (sender == lastSender) && !lastSender.isEmpty();
+    // messageType should be Event_RoomSay::UserMessage though we don't actually check
+    bool isUserMessage = !(userName.toLower() == "servatrice" || userName.isEmpty());
+    bool sameSender = isUserMessage && userName == lastSender;
     QTextCursor cursor = prepareBlock(sameSender);
-    lastSender = sender;
+    lastSender = userName;
 
     // timestamp
-    if (showTimestamps && (!sameSender || sender.toLower() == "servatrice") && !sender.isEmpty()) {
+    if (showTimestamps && ((!sameSender && isUserMessage) || userName.toLower() == "servatrice")) {
```

Comment:
better comments, i don't understand this

Reply:
it's better than the previous check